crypto/rand/rand_lib.c

```diff
@@ -211,25 +213,331 @@ void rand_cleanup_int(void)
     CRYPTO_THREAD_lock_free(rand_engine_lock);
 #endif
     CRYPTO_THREAD_lock_free(rand_meth_lock);
-    CRYPTO_THREAD_lock_free(rand_bytes.lock);
-    if (rand_bytes.secure)
-        OPENSSL_secure_clear_free(rand_bytes.buff, rand_bytes.size);
-    else
-        OPENSSL_clear_free(rand_bytes.buff, rand_bytes.size);
 }
 
 /*
- * RAND_poll_ex() gets a function pointer to call when it has random bytes.
- * RAND_poll() sets the function pointer to be a wrapper that calls RAND_add().
+ * RAND_poll() reseeds the default RNG using random input
+ *
+ * The random input is obtained from polling various entropy
+ * sources which depend on the operating system and are
+ * configurable via the --with-rand-seed configure option.
+ */
+int RAND_poll(void)
+{
+    int ret = 0;
+
+    RAND_POOL *pool = NULL;
+
+    const RAND_METHOD *meth = RAND_get_rand_method();
+
+    if (meth == RAND_OpenSSL()) {
+        /* fill random pool and seed the default DRBG */
+        RAND_DRBG *drbg = RAND_DRBG_get0_global();
+
+        if (drbg == NULL)
+            return 0;
+
+        CRYPTO_THREAD_write_lock(drbg->lock);
+        ret = rand_drbg_restart(drbg, NULL, 0, 0);
+        CRYPTO_THREAD_unlock(drbg->lock);
+
+        return ret;
+
+    } else {
+        /* fill random pool and seed the current legacy RNG */
+        pool = RAND_POOL_new(RAND_DRBG_STRENGTH,
+                             RAND_DRBG_STRENGTH/8,
+                             RAND_DRBG_STRENGTH/8);
+
+        if (RAND_POOL_acquire_entropy(pool) == 0)
+            goto err;
+
+        if (meth->add == NULL ||
+            meth->add(
+                RAND_POOL_buffer(pool),
+                RAND_POOL_length(pool),
+                (RAND_POOL_entropy(pool)/8.0)) == 0)
+            goto err;
+
+        ret = 1;
+    }
+
+err:
+    RAND_POOL_free(pool);
+    return ret;
+}
+
+/*
+ * The 'random pool' acts as a dumb container for collecting random
+ * input from various entropy sources. The pool has no knowledge about
+ * whether its randomness is fed into a legacy RAND_METHOD via RAND_add()
+ * or into a new style RAND_DRBG. It is the callers duty to 1) initialize the
+ * random pool, 2) pass it to the the polling callbacks, 3) seed the RNG,
```

Comment:
"the the" -> "the"

Reply:
Funny, looks like I'm not alone ;-)

```
~/src/openssl$ grep -r 'the the ' *
CHANGES:  *) Added more prefixes to parse for in the the strings written
CHANGES:  *) Add support for the the free Netwide assembler (NASM) under Win32,
Configurations/README:Configure places no rules on the command line, except the the first
crypto/des/asm/des_enc.m4:! After the the original LibDES initial permutation, the resulting left
crypto/des/asm/des_enc.m4:!  left and right ends up the the registers originally used.
crypto/rand/rand_lib.c: * random pool, 2) pass it to the the polling callbacks, 3) seed the RNG,
crypto/ec/ec_asn1.c:         * use the asn1 OID to describe the the elliptic curve parameters
doc/man3/SSL_SESSION_set1_id.pod:SSL_SESSION_set1_id() sets the the session ID for the B<ssl> SSL/TLS session
include/openssl/ec.h:/** Creates an ECPARAMETERS object for the the given EC_GROUP object.
include/openssl/ec.h:/** Creates an ECPKPARAMETERS object for the the given EC_GROUP object.
ssl/ssl_locl.h: * header as well as the the handshake message header.
test/ssl-tests/13-fragmentation.conf.in:    # around 4. Since the the test buffer is 64kB, a 4kB fragment is
```